### ISSUE TITLE
apple_library error @import module when mixing with other libraries

### ISSUE THREAD

**Issue by JuanFelippo** [ob/rules_ios#19]
Hi! i have 2 scenarios that fails:

## Scenario1: 

structure:
ModuleA (mixed, objc /swift)
-> depends on ModuleC (pure obc)

bazel build //MixedTest:ModuleA_swift

 error: umbrella header 'ModuleC-umbrella.h' not found
  umbrella header "ModuleC-umbrella.h"
##

## Scenario2:
structure:
ModuleD (pure swift)
-> depends on ModuleE (pure swift)

bazel build //MixedTest:ModuleD_swift

MixedTest/ModuleD/D.swift:2:8: error: cannot load underlying module for 'ModuleE'
import ModuleE

##

Fork to reproduce error:
https://github.com/JuanFelippo/rules_ios
branch: feature/mixed

Any ideasss?

**Comment by segiddins** [comment]
Try using `apple_framework` instead of `apple_library` ? `apple_library` is not meant to be used as a dependency by anything but packaging / bundling rules

**Comment by ob** [comment]
Yeah, what @segiddins said, also in your dependencies, don't depend on the `_objc` or `_swift` labels directly, just depend on `ModuleA`. I suspect you had to depend on those components _because_ you are using `apple_library` instead of `apple_framework`. 

If you make it work we should add it to the examples since we only have a basic one with a single `ios_application` rule.

**Comment by segiddins** [comment]
@ob maybe it makes sense to hide `apple_library` inside a struct so it can't be used from BUILD files?

**Comment by ob** [comment]
Yeah, let's do that.

**Comment by JuanFelippo** [comment]
I´ve just pushed a few changes using apple_framework 👍 
Good News! Scenario 2 was fixed.

About scenario 1, in order to use a swift class (from a framework) in a objc framework. What import should i import? 

#import "MixedTest/ModuleC-umbrella.h"  (not working)
#import "MixedTest/ModuleC_swift-Swift.h"  (not working)


+1 adding to examples if it works 😄

**Comment by segiddins** [comment]
Try `@import ModuleC;` maybe?

**Comment by JuanFelippo** [comment]
Excelent! It Worked!!!! 😄 !!!  Will make a PR to repo as example!
super clean @segiddins  @ob , Thanks so much for your help!

**Comment by edias** [comment]
Im having a similar situation where a method is added to a swift extension marked with `@objc` but the Objective-C class does not have visibility. `error: no visible @interface for 'Objective-C Class' declares the selector 'myMethod'(implemented on the swift extension).

I'm using a `apple_framework` and the code is mixed(ObjC + Swift). Any suggestion to solve this one?

**Comment by tinder-maxwellelliott** [comment]
> Yeah, let's do that.

I would be interested in how to do this refactor. Is there an example in the codebase of this already?
